src/main/ml-modules/services/geoQueryService.sjs

```diff
@@ -642,76 +642,86 @@ function getTimeBoundingWhereQuery(layerModel, req) {
   }
 }
 
+/**
+ *
+ * @param requestQuery the query from the caller's request
+ * @param layerModel
+ * @returns a CTS query
+ */
+function buildBoundingQuery(requestQuery, layerModel) {
+  const boundingQueries = [ parseGeometry(requestQuery, layerModel) ];
+
+  if (layerModel.boundingQuery) {
+    boundingQueries.push(qd.query(layerModel.boundingQuery));
+  }
+  if (layerModel.temporalBounds) {
+    boundingQueries.push(getTemporalQuery(layerModel.temporalBounds));
+  }
+
+  // alwaysIncludeQuery is an optional feature for a layer definition to ensure that certain data is always included,
+  // which implies usage of an "or" query. See https://github.com/marklogic-community/marklogic-geo-data-services/issues/76
+  // for the original requirements.
+  const boundingQuery = layerModel.alwaysIncludeQuery ?
+    cts.orQuery([qd.query(layerModel.alwaysIncludeQuery), cts.andQuery(boundingQueries)]) :
+    cts.andQuery(boundingQueries);
+
+  xdmp.trace("GDS-DEBUG", "boundingQuery: " + xdmp.toJsonString(boundingQuery));
+  return boundingQuery;
+}
+
 // returns a Sequence of documents
 function getObjects(req, exportPlan=false) {
 
   xdmp.trace("GDS-DEBUG", "Starting getObjects");
   xdmp.trace("GDS-DEBUG", "getLayerModel(" + req.params.id + ", " + req.params.layer + ")" );
   const layerModel = sm.getLayerModel(req.params.id, req.params.layer);
 
-  const query = req.query;
-  const orderByFields = parseOrderByFields(query);
-  const geoQuery = parseGeometry(query, layerModel);
+  const requestQuery = req.query;
```

Comment:
Renamed `query` to `requestQuery` since there are multiple query objects flying around here, and I wanted to make it obvious what `query` actually refers to.